panther_hardware_interfaces/test/test_gpio_controller.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+
+#include <gtest/gtest.h>
+#include <gpiod.hpp>
+
+#include <panther_hardware_interfaces/gpio_controller.hpp>
+
+class GPIOControllerWrapper : public panther_hardware_interfaces::GPIOControllerPTH12X
+{
+public:
+  void WatchdogEnable() { this->watchdog_->TurnOn(); }
+  void WatchdogDisable() { this->watchdog_->TurnOff(); }
+  bool IsWatchdogEnabled() { return this->watchdog_->IsWatchdogEnabled(); }
+};
+
+class TestGPIOController : public ::testing::Test
+{
+public:
+  void SetUp() override;
+  void TearDown() override { gpio_controller_wrapper_.reset(); }
```

Comment:
Is there any specific reason for using the `SetUp` and `TearDown` methods? The general approach suggests using constructor/destructor except. Despite the GPIO tests, we use the const/dest approach in the Panther project.

https://fuchsia.googlesource.com/third_party/googletest/+/HEAD/googletest/docs/faq.md#ctorvssetup